dev/MalloyExplorer.tsx

```diff
@@ -41,43 +43,55 @@ const App = () => {
       >
         <ExplorerPanelsContext.Provider
           value={{
-            isSourcePanelOpen,
             setIsSourcePanelOpen,
+            setIsQueryPanelOpen,
           }}
         >
           <div {...stylex.props(styles.page)}>
             <div {...stylex.props(styles.content)}>
-              {isSourcePanelOpen && (
+              {isSourcePanelOpen ? (
                 <div
                   {...stylex.props(styles.panel)}
                   style={{width: `${sourcePanelWidth}px`}}
                 >
-                  <SourcePanel />
+                  <SourcePanel onRefresh={() => {}} />
                   <ResizeBar
                     minWidth={180}
                     width={sourcePanelWidth}
                     onWidthChange={setSourcePanelWidth}
                   />
                 </div>
-              )}
-
-              <div
-                {...stylex.props(styles.panel)}
-                style={{width: `${queryPanelWidth}px`}}
-              >
-                <QueryPanel
-                  runQuery={(source, query) => {
-                    const qb = new QueryBuilder.ASTQuery({source, query});
-                    window.alert(qb.toMalloy());
-                  }}
+              ) : (
+                <CollapsedPagePanel
+                  icon="database"
+                  title={source.name}
+                  onExpand={() => setIsSourcePanelOpen(true)}
                 />
-                <ResizeBar
-                  minWidth={230}
-                  width={queryPanelWidth}
-                  onWidthChange={setQueryPanelWidth}
+              )}
+              {isQueryPanelOpen ? (
+                <div
+                  {...stylex.props(styles.panel)}
+                  style={{width: `${queryPanelWidth}px`}}
+                >
+                  <QueryPanel
+                    runQuery={(source, query) => {
+                      const qb = new QueryBuilder.ASTQuery({source, query});
+                      window.alert(qb.toMalloy());
+                    }}
+                  />
+                  <ResizeBar
+                    minWidth={280}
+                    width={queryPanelWidth}
+                    onWidthChange={setQueryPanelWidth}
+                  />
+                </div>
+              ) : (
+                <CollapsedPagePanel
+                  icon="filterSliders"
+                  title="Query"
+                  onExpand={() => setIsQueryPanelOpen(true)}
                 />
-              </div>
-
+              )}
```

Comment:
I think that there's enough duplicated code here to perhaps roll into a component so that you can do something like:
```tsx
<ResizeableCollapseablePanel 
  minWidth={280}
  width={queryPanelWidth}
  onWidthChange={setQueryPanelWidth}
>
  <QueryPanel
    runQuery={(source, query) => {
      const qb = new QueryBuilder.ASTQuery({source, query});
      window.alert(qb.toMalloy());
    }}
  />
</ResizeableCollapseablePanel>
```
Also, you probably don't want to conditionally render the inner component, because that will drop all of the inner state, including expanded states of nests. You probably want to `display: none` the whole component so that it rehydrates quickly and in the state it was before it closed.